### ISSUE TITLE
Failed to import contact

### ISSUE THREAD

**Issue by LantashTheTokra** [SimpleMobileTools/Simple-Contacts#1047]
When I try to open this contact from the file manager, it fails to import. The GrapheneOS contacts app imports it right, so it's not an issue with the file
App version: 6.22.7
Android version: 14
Distro: GrapheneOS
File: https://github.com/LantashTheTokra/smsbug
Screenshot with error message:
![Screenshot_20231017-122652](https://github.com/SimpleMobileTools/Simple-Contacts/assets/84555704/4935cef7-510c-4c88-b382-9a8023638844)

**Comment by hetszunyukapanyanyimonyok** [comment]
I have the same problem. It sucks because reseted phone and no contacts.

**Comment by hetszunyukapanyanyimonyok** [comment]
I had automatic backup, everyone of them says importing failed Additional information: I tried with older versions too and those not work too. Maybe the exports faulty?

**Comment by hetszunyukapanyanyimonyok** [comment]
Tested with and older .vcf file (was exported with simple contact, and hidden numbers)and the import worked ! None of the new automated backup worked.

**Comment by Danfro** [comment]
Same error here when trying to import an vcf that was exported from Android. Version 6.22.7. pro

**Comment by Danfro** [comment]
Did some try&error editing the vcf file. When removing the following entry for all contacts import worked: `SOUND;X-IRMC-N:;;;;`.

**Comment by Danfro** [comment]
> When I try to open this contact from the file manager, it fails to import. The GrapheneOS contacts app imports it right, so it's not an issue with the file App version: 6.22.7 Android version: 14 Distro: GrapheneOS File: https://github.com/LantashTheTokra/smsbug/blob/main/20231016_232659_01.vcf Screenshot with error message: ![Screenshot_20231017-122652](https://user-images.githubusercontent.com/84555704/275914552-4935cef7-510c-4c88-b382-9a8023638844.png)

Btw. I removed the LOGO part from your vcf. Then import works.

**Comment by hetszunyukapanyanyimonyok** [comment]
@Danfro what did you use for edit?

**Comment by Danfro** [comment]
@hetszunyukapanyanyimonyok Just Ubuntu default Text editor. That allowed to search&replace including the line break. Nothing fancy. Most editors will do I suppose.

**Comment by Danfro** [comment]
I am sure "fixing" this is almost impossible. vcf is not vcf. There are several versions of the standard and many OS or apps do implement their own addons or modifications. Implementing them all in this app would be a huge task.

BUT a proper error message dialog that explains the reason (not supported tags/not standard conform tags) would be really helpful. And ideally a list of those tags which fail would be perfect.

**Comment by hetszunyukapanyanyimonyok** [comment]
So I tried @Danfro solution with the
SOUND;X-IRMC-N:;;;;
there is none any of these in my vcf and tried to find and remove LOGO could not find. So this import bug must be something unique? Can't find anything useful. Anybody found any other solution?

**Comment by Danfro** [comment]
> So I tried @Danfro solution with the SOUND;X-IRMC-N:;;;; there is none any of these in my vcf and tried to find and remove LOGO could not find. So this import bug must be something unique? Can't find anything useful. Anybody found any other solution?

Exactly. It can by anything and something else for everyone. That's why fixing it is not easy.

You can only try this:
1. make a copy of your file
2. delete all except one contact
3. try to load that file
4. if it fails: remove the lines one by one, start with them using your intuition
5. if it works: remove that contact and add some more contacts, do step 4 as long as it works
6. delete the "bad code" for all other contacts where that occurs
7. try the whole file
8. if it still breaks, back to step 4

That is what I did.

**Comment by hetszunyukapanyanyimonyok** [comment]
@Danfro Thank for the methodology! 
I will test it in the afternoon. I hope it will work. 
@Danfro After fixing the file, did the automatic backup work(again as an importable vcf)?

**Comment by Danfro** [comment]
@Danfro After fixing the file, did the automatic backup work(again as an importable vcf)?

I did not try to import a backup, but a new file. So I can't tell you. But should work... Good luck.

**Comment by Danfro** [comment]
Another option to get contacts imported for Thunderbird users, seems to be to install the [CardBook extension](https://addons.thunderbird.net/thunderbird/addon/cardbook/). This apparently allows to get properly formatted vcards. Those can then be exported to a vcf, which then can be imported in contacts app.
